src/main/java/com/example/Restaurant_Management_System_REST_API/service/TableService.java

```diff
@@ -0,0 +1,103 @@
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class TableService {
+
+    private final TableRepository tableRepository;
+    private final ModelMapper modelMapper;
+
+    public TableDTO add(TableDTO tableDTO) {
+        Table table = modelMapper.map(tableDTO, Table.class);
+        tableRepository.save(table);
+
+        return modelMapper.map(table, TableDTO.class);
+    }
+
+    public List<TableDTO> findAll() {
+        List<TableDTO> tableList = new ArrayList<>();
+        tableRepository.findAll().forEach(table -> tableList.add(modelMapper.map(table, TableDTO.class)));
+
+        return tableList;
+    }
+
+    private Table findById(Long id) throws NotFoundInDatabaseException {
+        return tableRepository.findById(id).orElseThrow(() -> new NotFoundInDatabaseException(Table.class));
+    }
+
+    public ResponseEntity<?> deleteById(Long id) throws NotFoundInDatabaseException {
+        Table tableToDelete = tableRepository.findById(id).orElseThrow(() -> new NotFoundInDatabaseException(Table.class));
+
+        tableRepository.delete(tableToDelete);
+
+        return new ResponseEntity<>("Table with id " + tableToDelete.getId() + " has been deleted!", HttpStatus.OK);
+    }
+
+    void iterateAndSetTablesToReservationAndSave(Reservation reservation) throws NotFoundInDatabaseException {
+        for (Table table : reservation.getTables()) {
+            checkIfTableExist(table.getId());
+            List<Reservation> reservationList = new ArrayList<>();
+
+            if(table.getReservationList() != null)
+                reservationList = table.getReservationList();
+
+            reservationList.add(reservation);
+            table.setReservationList(reservationList);
+
+            table.setAvailable(false);
+            tableRepository.save(table);
+        }
+    }
+
+    private void checkIfTableExist(Long id) throws NotFoundInDatabaseException {
+        tableRepository.findById(id).orElseThrow(() -> new NotFoundInDatabaseException(Table.class));
+    }
+
+    void iterateAndSetReservationToNullInTablesAndSave(Reservation reservationToBeDeleted) {
+        for (Table table : reservationToBeDeleted.getTables()) {
+            table.setReservationList(null);
+            table.setAvailable(true);
+            tableRepository.save(table);
+        }
+    }
+
+    void checkIfTablesAreAvailable(Reservation reservation) throws NotFoundInDatabaseException {
+        for (Table table : reservation.getTables()) {
+            Table existingTable = findById(table.getId());
+
+            if (existingTable != null) {
+                for (Reservation existingReservation : existingTable.getReservationList()) {
+                    if (isTimeConflict(existingReservation.getStart(), reservation.getStart())) {
+                        throw new ResponseStatusException(HttpStatus.CONFLICT, "Table with id " + table.getId() +
+                                " is not available at the requested time.");
```

Comment:
To improve code readability here it can be extracted to private methods so it'll be more atomic.

Reply:
okay! maybe like this?

![image](https://github.com/BartoliniAndBorderCollies/Restaurant_Management_System_REST_API/assets/126821059/a51703ac-7318-4231-aaad-ea40137206f7)